EIPS/eip-3085.md

```diff
@@ -48,16 +48,29 @@ The wallet **MAY** reject the request for any reason.
 `wallet_addEthereumChain` accepts a single object parameter, specified by the following TypeScript interface:
 
 ```typescript
-interface AddEthereumChainParameter {
+interface BaseNativeCurrency {
+  name: string;
+  symbol: string;
+}
+
+interface NativeCurrencyWithDecimals extends BaseNativeCurrency {
+  decimals: number;
+}
+
+interface NativeCurrencyWithScalingFactor extends BaseNativeCurrency {
+  scalingFactor: string;
+}
+
+type NativeCurrency =
+  | NativeCurrencyWithDecimals
+  | NativeCurrencyWithScalingFactor;
+
+interface AddEthereumChainParameter<T extends NativeCurrency> {
```

Comment:
This feels unnecessarily complicated since `decimals` can be represented as a `scalingFactor`.  e.g., `decimals: 6` is the same as `scalingFactor: '0xf4240'`.